app/web/src/newhotness/ComponentContextMenu.vue

```diff
@@ -35,6 +107,57 @@ const rightClickMenuItems = computed(() => {
     onSelect: () => {},
   });
 
+  // Only enable actions if we are working with a single component.
+  if (id.value && schemaVariantId.value) {
+    const submenuItems: DropdownMenuItemObjectDef[] = [];
+
+    for (const actionPrototype of actionPrototypes.value) {
+      const existingActionId = actionByPrototype.value[actionPrototype.id];
+      submenuItems.push({
+        label: actionPrototype.displayName || actionPrototype.name,
+        toggleIcon: true,
```

Comment:
This _should_ be preventing menu close... we'll live for this PR to keep moving as many will be touching the context menu and refining it